src/ashtha/Shop.java

```diff
@@ -0,0 +1,121 @@
+package ashtha;
+
+public class Shop {
+
+	int initialMaggieStocks;
+	int initialDosaStocks;
+	int initialPouchesStocks;
+	int initialPanipuriStocks;
+	int initialMasalaStocks;
+	
+	/*method to set the inventory values*/
+	void setInventory (int maggieStock, int dosaStock, int pouchesStock, int panipuriStock, int masalaStock) {
+		initialMaggieStocks = maggieStock;
+		initialDosaStocks = dosaStock;
+		initialPouchesStocks = pouchesStock;
+		initialPanipuriStocks = panipuriStock;
+		initialMasalaStocks = masalaStock;
+	}
+	
+	/*method to update the inventory value*/
+	void updateInventory (int requestedMaggieQty, int requestedDosaQty, int requestedPouchesQty, int requestedPanipuriQty, int requestedMasalaQty, boolean partial) {
+		/* to update maggie stock*/
+		if (initialMaggieStocks >= requestedMaggieQty) {
+			initialMaggieStocks = initialMaggieStocks - requestedMaggieQty;
+		}else if (initialMaggieStocks == 0) {
+			System.out.println("Maggie is currently out of stock");
+		}else if (true) {
+			System.out.println("You requested for "+requestedMaggieQty +" maggie packets but only "+initialMaggieStocks+" packets available in stock. Do you want to proceed with the cart?");
+			System.out.println("Order placed successfully for "+initialMaggieStocks +" maggie packets.");
+			initialMaggieStocks = 0;
+		}
+		/*to update Dosa stock*/
+		if (initialDosaStocks >= requestedDosaQty) {
+			initialDosaStocks = initialDosaStocks - requestedDosaQty;
+		}else if (initialDosaStocks == 0) {
+			System.out.println("Dosa is currently out of stock");
+		}else if (true) {
+			System.out.println("You requested for "+requestedDosaQty +" dosa but only "+initialDosaStocks+" available in stock. Do you want to proceed with the cart?");
+			System.out.println("Order placed successfully for "+initialDosaStocks +" dosa.");
+			initialDosaStocks = 0;
+		}
+		/*to update Pouches stock*/
+		if (initialPouchesStocks >= requestedPouchesQty) {
+			initialPouchesStocks = initialPouchesStocks - requestedPouchesQty;
+		}else if (initialPouchesStocks == 0) {
+			System.out.println("Pouches are currently out of stock");
```

Comment:
Why we printing the OOS statements here, we have separate requirement for this FR2 and FR3